helm/values.yaml

```diff
@@ -65,15 +65,43 @@ coder:
 
   # coder.resources -- The resources to request for Coder. These are optional
   # and are not set by default.
-  resources:
-    {}
+  resources: {}
     # limits:
     #   cpu: 100m
     #   memory: 128Mi
     # requests:
     #   cpu: 100m
     #   memory: 128Mi
 
+  # coder.affinity -- Allows specifying an affinity rule for the `coder` deployment.
+  # The default rule prefers to schedule coder pods on different
+  # nodes, which is only applicable if coder.replicaCount is greater than 1.
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+        - podAffinityTerm:
+            labelSelector:
+              matchExpressions:
+                - key: app.kubernetes.io/name
+                  operator: In
+                  values:
+                    - "coder"
```

Comment:
This will not work in all cases because `app.kubernetes.io/name` is set to the template `coder.name` in `_helpers.tpl`, which is based on the release name.

You can't use variables in values.yaml, so we can't resolve this by including the template sadly. I think the best way forward would be to add a new label to the `coder.labels` block that is static such as `app.coder.com: true` or something